recipe/bld.bat

```diff
@@ -1,47 +1,221 @@
 @echo On
+setlocal enabledelayedexpansion
+
+REM remove pyproject.toml to avoid installing deps from pip
+if EXIST pyproject.toml DEL pyproject.toml
 
 set TH_BINARY_BUILD=1
 set PYTORCH_BUILD_VERSION=%PKG_VERSION%
-set PYTORCH_BUILD_NUMBER=%PKG_BUILDNUM%
+:: Always pass 0 to avoid appending ".post" to version string.
+:: https://github.com/conda-forge/pytorch-cpu-feedstock/issues/315
+set PYTORCH_BUILD_NUMBER=0
+
+REM I don't know where this folder comes from, but it's interfering with the build in osx-64
+if EXIST %PREFIX%\git RD /S /Q %PREFIX%\git
 
-if "%pytorch_variant%" == "gpu" (
-    set build_with_cuda=1
-    set desired_cuda=%CUDA_VERSION:~0,-1%.%CUDA_VERSION:~-1,1%
+@REM Setup BLAS
+if "%blas_impl%" == "generic" (
+    REM Fake openblas
+    SET BLAS=OpenBLAS
+    SET OpenBLAS_HOME=%LIBRARY_PREFIX%
 ) else (
-    set build_with_cuda=
-    set USE_CUDA=0
+    SET BLAS=MKL
 )
 
-if "%build_with_cuda%" == "" goto cuda_flags_end
+@REM TODO(baszalmstra): Figure out if we need these flags
+SET "USE_NUMA=0"
+SET "USE_ITT=0"
 
-set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%desired_cuda%
-set CUDA_BIN_PATH=%CUDA_PATH%\bin
-set TORCH_CUDA_ARCH_LIST=5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX
-set TORCH_NVCC_FLAGS=-Xfatbin -compress-all
+@REM KINETO seems to require CUPTI and will look quite hard for it.
+@REM CUPTI seems to cause trouble when users install a version of
+@REM cudatoolkit different than the one specified at compile time.
+@REM https://github.com/conda-forge/pytorch-cpu-feedstock/issues/135
+set "USE_KINETO=OFF"
 
-:cuda_flags_end
+if "%PKG_NAME%" == "pytorch" (
+  set "PIP_ACTION=install"
+  :: We build libtorch for a specific python version.
+  :: This ensures its only build once. However, when that version changes
+  :: we need to make sure to update that here.
+  :: Get the full python version string
+  for /f "tokens=2" %%a in ('python --version 2^>^&1') do set PY_VERSION_FULL=%%a
 
-set DISTUTILS_USE_SDK=1
+  :: Replace Python312 or python312 with ie Python311 or python311
+  sed "s/\([Pp]ython\)312/\1%CONDA_PY%/g" build/CMakeCache.txt.orig > build/CMakeCache.txt
 
-set CMAKE_INCLUDE_PATH=%LIBRARY_PREFIX%\include
-set LIB=%LIBRARY_PREFIX%\lib;%LIB%
+  :: Replace version string v3.12.8() with ie v3.11.11()
+  sed -i.bak -E "s/v3\.12\.[0-9]+/v%PY_VERSION_FULL%/g" build/CMakeCache.txt
```

Comment:
fwiw we're using the builtin `BUILD_LIBTORCH_WHL` variables so we don't have to do all this. Seems to be working and testing well although I haven't checked in detail. https://github.com/AnacondaRecipes/pytorch-feedstock/blob/58b281e091816c0074e3754b31a8596f4e2ddaad/recipe/bld.bat#L111
If this is working, though, might be worth merging and looking at this later. I don't want to de-rail it when it's almost over the line. @ELundby45

Reply:
At least this didn't seem to work on Windows (hence sccache), but on Linux the difference is that doing the `sed` causes CMake to reuse most of the compiled files, but using re-cmake means it rebuilds a lot of common bits.

Reply:
I’m not sure if we really ever tried their inbuilt approach - I haven’t but maybe @baszalmstra has?

Reply:
Nope, my attempts were all before BUILD_LIBTORCH_WHL was available. If it works without this “hack” it would be a big improvement because we can also get rid of the sccache workaround.

Reply:
Well, we have time to experiment a bit, currently the open-gpu-server is completely down and won't come back up before next week. So if someone wants to push into this PR (or you can try it out in #231 and I merge it here afterwards), feel free.

Reply:
The upstream options were discussed in #272. Long story short, I've tried and it was worse than status quo (the whole ATen tree was built multiple times).

Reply:
yep ok, we're seeing similar things. thanks for the input. Ideally this mechanism could be fixed upstream, probably.